plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways.tsx

```diff
@@ -44,23 +45,41 @@ export const PaymentGateways = () => {
 		);
 	}, [] );
 
-	const recommendedGateways = [ 'woocommerce_payments' ];
-
 	const setupLivePayments = () => {
 		// TODO: Implement in future PR.
 	};
 
+	const determineGatewayStatus = ( gateway: PaymentGateway ) => {
+		if ( gateway.enabled ) {
+			if ( gateway.needs_setup ?? false ) {
+				return 'needs_setup';
+			}
+
+			if ( gateway.id === 'woocommerce_payments' ) {
+				if ( wooPaymentsGatewayData?.isInTestMode ) {
+					return 'test_mode';
+				}
+			}
+			return 'active';
+		}
+
+		if ( gateway.id === 'woocommerce_payments' ) {
+			return 'recommended';
+		}
+
+		return 'inactive';
+	};
+
 	// Transform plugins comply with List component format.
 	const paymentGatewaysList = paymentGateways.map(
```

Comment:
Just a thought. It may be worth moving this transformation logic into a `useMemo` hook to prevent unnecessary recalculations on re-renders.